src/ITensorChainRules/ITensorChainRules.jl

```diff
@@ -364,4 +410,32 @@ function ChainRulesCore.rrule(::typeof(inner), x1::MPS, x2::MPS; kwargs...)
   return y, inner_pullback
 end
 
+function ChainRulesCore.rrule(::typeof(*), x1::MPO, x2::MPO; kwargs...)
+  y = *(x1, x2; kwargs...)
+  function contract_pullback(ȳ)
+    x̄1 = *(ȳ, dag(x2); kwargs...)
+    x̄2 = *(dag(x1), ȳ; kwargs...)
+    return (NoTangent(), x̄1, x̄2)
+  end
+  return y, contract_pullback
+end
+
+function ChainRulesCore.rrule(::typeof(tr), x::MPO; kwargs...)
+  y = tr(x; kwargs...)
+  function contract_pullback(ȳ)
+    s = firstsiteinds(x)
+    #return ȳ * MPO(s, "Id")
```

Comment:
What's the problem with `MPO(s, "Id")`? This works for me:
```julia
julia> using ITensors

julia> s = siteinds("S=1/2", 4);

julia> I = MPO(s, "Id");

julia> I' * I
MPO
[1] ((dim=2|id=573|"S=1/2,Site,n=1")'', (dim=2|id=573|"S=1/2,Site,n=1"), (dim=1|id=198|"Link,l=1"))
[2] ((dim=2|id=104|"S=1/2,Site,n=2")'', (dim=2|id=104|"S=1/2,Site,n=2"), (dim=1|id=709|"Link,l=2"), (dim=1|id=198|"Link,l=1"))
[3] ((dim=2|id=82|"S=1/2,Site,n=3")'', (dim=2|id=82|"S=1/2,Site,n=3"), (dim=1|id=574|"Link,l=3"), (dim=1|id=709|"Link,l=2"))
[4] ((dim=2|id=97|"S=1/2,Site,n=4")'', (dim=2|id=97|"S=1/2,Site,n=4"), (dim=1|id=574|"Link,l=3"))
```

Reply:
Yes, there was an error that I mistakenly attributed to this. Sometimes it is tricky with Zygote when you have multiple bugs.